From a60d1c3dc250341cdd8b1577941e2f4a4e8beb6b Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Fri, 26 Apr 2024 07:56:11 +0900
Subject: [PATCH 5/5] Remove unused broken clone function

The base class doesn't have a clone member function, so the code was
never valid. As it was unused, that didn't cause problems until clang
caught this earlier than instantiation in
https://github.com/llvm/llvm-project/pull/84050
---
 cctools/ld64/src/ld/code-sign-blobs/blob.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cctools/ld64/src/ld/code-sign-blobs/blob.h b/cctools/ld64/src/ld/code-sign-blobs/blob.h
index 54afb2d..c17d19b 100644
--- a/cctools/ld64/src/ld/code-sign-blobs/blob.h
+++ b/cctools/ld64/src/ld/code-sign-blobs/blob.h
@@ -181,9 +181,6 @@ public:
 		return NULL;
 	}
 	
-	BlobType *clone() const
-	{ assert(validateBlob()); return specific(this->BlobCore::clone());	}
-
 	static BlobType *readBlob(int fd)
 	{ return specific(BlobCore::readBlob(fd, _magic, sizeof(BlobType), 0), true); }
 
